packages/ibm-products/src/components/NotificationsPanel/NotificationsPanel.stories.jsx

```diff
@@ -45,149 +56,206 @@ export default {
   },
 };
 
-const renderUIShellHeader = (
-  open,
-  setOpen,
-  notificationTriggerRef,
-  hasUnreadNotifications
-) => (
-  <HeaderContainer
-    render={() => (
+const defaultProps = {
+  data: 0,
+  open: true,
+  onDoNotDisturbChange: action('Toggled "Do not disturb"'),
+  onViewAllClick: action('Clicked "View all"'),
+  onSettingsClick: action('Clicked gear icon'),
+};
+
+// Create a new notification.
+const newNotification = () => {
+  // Pick a random notification from the existing data set.
+  const random = Math.floor(Math.random() * data.length);
+  const notification = { ...data[random] };
+  // Update dynamic data.
+  notification.id = uuidv4();
+  notification.timestamp = new Date();
+  notification.unread = true;
+
+  return notification;
+};
+
+const Template = (args) => {
+  const { data, open, ...rest } = args;
+  const [notificationsData, setNotificationsData] = useState(data);
+  const [hasUnreadNotifications, setHasUnreadNotifications] = useState(true);
+  const [userOpen, setUserOpen] = useState(false);
+  const [notificationsOpen, setNotificationsOpen] = useState(open);
+  const [switcherOpen, setSwitcherOpen] = useState(false);
+  const userActionRef = useRef(null);
+  const notificationActionRef = useRef(null);
+  const switcherActionRef = useRef(null);
+
+  const markAllAsUnread = () => {
+    let tempData = [...notificationsData];
+    tempData.forEach((element) => (element.unread = false));
+    setNotificationsData(tempData);
+  };
+
+  // Every time data is added or removed, check for unread notifications.
+  useEffect(() => {
+    const hasUnread = notificationsData.find(
+      (notification) => notification.unread === true
+    );
+    setHasUnreadNotifications(hasUnread);
+  }, [notificationsData]);
+
+  // Every time the panel is opened, mark all notifications as unread.
+  useEffect(() => {
+    if (notificationsOpen) {
+      markAllAsUnread();
+    }
+  }, [notificationsOpen]);
+
+  // After changing `data` from the Storybook control.
+  useEffect(() => {
+    setNotificationsData(data);
+  }, [data]);
+  // After changing `open` from the Storybook control.
+  useEffect(() => {
+    setNotificationsOpen(open);
+  }, [open]);
+
+  const addNewNotification = () => {
+    const notification = newNotification();
+    setNotificationsData((data) => [...data, { ...notification }]);
+  };
+
+  return (
+    <div className={`${storyBlockClass}--full-height`}>
+      {/* <HeaderContainer
+        render={() => ( */}
```

Comment:
Temp fix for draft review.

Removing the `HeaderContainer`:
1. Solves the problem of re-rendering all children every time there's a state change. 
2. Solves the problem of tabbing order by allowing the Panel to be rendered inline with the Shell's notification icon button.

Keeping the `HeaderContainer`:
1. Requires the Panel to be rendered outside the Shell, but it will lose the expected tabbing order.